src/index.js

```diff
@@ -24,39 +24,44 @@ export default class Modal extends Component {
       if(this.props && this.props.styles){
         Object.keys(this.props.styles).forEach((key)=>{
           style[effect].panel[key] = this.props.styles[key];
-          if(key === 'width'){
-            if(this.props.styles[key].charAt(this.props.styles[key].length-1) === '%'){
-              const width = this.props.styles[key].slice(0,-1);
-              style[effect].panel.marginLeft = '-' + width / 2 + '%';
-            }else if(this.props.styles[key].charAt(this.props.styles[key].length-1) === 'x'){
-              const width = this.props.styles[key].slice(0,-2);
-              style[effect].panel.marginLeft = '-' + width / 2 + 'px';
-            }else{
-              style[effect].panel.marginLeft = '-' + width / 2 + 'px';
-            }
-          }else if(key === 'height'){
-            if(this.props.styles[key].charAt(this.props.styles[key].length-1) === '%'){
-              const height = this.props.styles[key].slice(0,-1);
-              style[effect].panel.marginTop = '-' + height / 2 + '%';
-            }else if(this.props.styles[key].charAt(this.props.styles[key].length-1) === 'x'){
-              const height = this.props.styles[key].slice(0,-2);
-              style[effect].panel.marginTop = '-' + height / 2 + 'px';
-            }else{
-              style[effect].panel.marginTop = '-' + height / 2 + 'px';
-            }
-          }
         });
       }
     }
 
     setSize(effect) {
         if(this.props && this.props.width) {
+          if(this.props.width.charAt(this.props.width.length-1) === '%'){
+            //Use Percentage
+            const width = this.props.width.slice(0,-1);
+            style[effect].panel.width = width + '%';
+            style[effect].panel.marginLeft = '-' + width / 2 + '%';
+          }else if(this.props.width.charAt(this.props.width.length-1) === 'x'){
+            //Use Pixels
+            const width = this.props.width.slice(0,-2);
+            style[effect].panel.width = width + 'px';
+            style[effect].panel.marginLeft = '-' + width / 2 + 'px';
+          }else{
+            //Default
             style[effect].panel.width = this.props.width + 'px';
             style[effect].panel.marginLeft = '-' + this.props.width / 2 + 'px';
+          }
         }
         if(this.props && this.props.height) {
+          if(this.props.height.charAt(this.props.height.length-1) === '%'){
+            //Use Percentage
+            const height = this.props.height.slice(0,-1);
+            style[effect].panel.height = height + '%';
+            style[effect].panel.marginTop = '-' + height / 2 + '%';
```

Comment:
This is a screen capture of the example page.
<img width="1042" alt="2016-11-20 22 41 39" src="https://cloud.githubusercontent.com/assets/4659294/20463232/ba38b192-af72-11e6-8eba-22e7833200b9.png">

The vertical position is a little bit shifted.
In this case, you had better use 'vh' like this.
```
style[effect].panel.height = height + 'vh';
style[effect].panel.marginTop = '-' + height / 2 + 'vh';
```
Please confirm it.